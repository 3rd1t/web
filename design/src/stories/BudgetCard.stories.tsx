import React from "react";
import { BudgetCard } from '../components/BudgetCard';

export default {
  title: "Budget Card",
};

export const card = () => {

  return (
    <BudgetCard  expensesCount= {195} totalBudget={1500.00} start={new Date()} end={new Date()} budgeted={4.99} expenses={625.12} income={10.00}/>
  )
};
