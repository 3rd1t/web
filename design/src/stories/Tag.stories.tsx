import React from 'react';
import Tag from "../components/Tag"
export default { title: 'Tag' };

export const Default = () => {
    return (
        <Tag>Hello Tag</Tag>
    )
} 
