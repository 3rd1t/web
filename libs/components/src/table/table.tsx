import React from "react";
import { Header } from "./header/header";
import { Preview } from "./cells/preview/preview";

export interface TableProps {
  columnNames: string[];
  rows: React.ReactNode[];
}

export const Table = ({ columnNames, rows }: TableProps) => {
  console.log({rows})
  const usedRows =
    rows.length > 0
      ? rows
      : [1, 2, 3].map((_) => {
          return (
            <tr>
              {columnNames.map((_) => (
                <Preview />
              ))}
            </tr>
          );
        });
  return (
    <div className="flex flex-col ">
      <div className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
        <div className="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8">
          <div className="overflow-hidden border border-gray-200 sm:rounded">
            <table className="min-w-full ">
              {<Header labels={columnNames} />}
              <tbody className="bg-white divide-y divide-gray-200">
                {usedRows}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Table;
