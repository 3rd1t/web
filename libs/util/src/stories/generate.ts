/**
 * Return all components from storybook.
 *
 * @example
 * import * as stories from "./<component>.stories"
 *
 * const testCases = generateTestCasesFromStories(stories)
 * testCases.forEach((s) => {
 *   it(s.name + "should match snapshot", () => {
 *     const { baseElement } = render(s.component())
 *     expect(baseElement).toMatchSnapshot()
 *   })
 * })
 *
 *
 * @param stories - import * as stories from "./<component>.stories"
 * @returns Name and render function for each component
 */
export function generateTestCasesFromStories(
  stories: Record<string, any>, // eslint-disable-line
): { name: string; component: () => HTMLElement }[] {
  return Object.entries(stories)
    .filter((s) => s[0] !== "default")
    .map((s) => {
      return {
        name: s[0],
        component: s[1],
      }
    })
}
