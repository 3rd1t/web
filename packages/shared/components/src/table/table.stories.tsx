import React from "react"
import { Table } from "./table"
import { Row } from "./row/row"
import { Header } from "./header/header"
import { Simple } from "./cells/simple/simple"
import { Multiline } from "./cells/multiline/multiline"
import { Tag } from "./cells/tag/tag"
import { action } from "@storybook/addon-actions"

import { Icon } from "./cells/icon/icon"
import { Button } from "./cells/button/button"

export default {
  component: Table,
  title: "components/table",
  subcomponents: { Row, Header },
}

export const singleRow = () => {
  const columnNames = ["Name", "Number", "Updated at"]
  const rows = [
    [<Simple label="Andreas Thomas" />, <Tag label={123456789} color="purple" />, <Simple label="22.09.2020" />],
  ]

  return <Table columnNames={columnNames} rows={rows} />
}

export const multipleRows = () => {
  const columnNames = ["Name", "Number", "Updated at"]

  const rows = [
    [<Simple label="Andreas Thomas" />, <Tag label={123456789} color="purple" />, <Simple label="22.09.2020" />],
    [
      <Multiline title="Nico Webersinke" content="Product lead" />,
      <Simple label={123456789} />,
      <Simple label="22.09.2020" />,
    ],
    [<Simple label="Mads Jordt" />, <Simple label={123456789} />, <Simple label="22.09.2020" />],
    [<Simple label="Kevin Kohler" />, <Simple label={123456789} />, <Simple label="22.09.2020" />],
  ]

  return <Table columnNames={columnNames} rows={rows} />
}

export const onDark = () => {
  const columnNames = ["Name", "Number", "Updated at"]

  const rows = [
    [<Simple label="Andreas Thomas" />, <Tag label={123456789} color="purple" />, <Simple label="22.09.2020" />],
    [
      <Multiline title="Nico Webersinke" content="Product lead" />,
      <Simple label={123456789} />,
      <Icon
        icon={
          <svg
            className="w-full"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
              d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12"
            />
          </svg>
        }
        color="red"
      />,
    ],
    [
      <Simple label="Mads Jordt" />,
      <Multiline title="Lorem" content="Lorem ipsum dolor sit amet, consectetur adipiscing elit." />,
      <Simple label="22.09.2020" />,
    ],
    [
      <Simple label="Kevin Kohler" />,
      <Simple label={123456789} />,
      <Button label="Click me" onClick={action("Clicked")} />,
    ],
  ]

  return (
    <div className="p-10 bg-gray-900">
      <Table columnNames={columnNames} rows={rows} />
    </div>
  )
}
