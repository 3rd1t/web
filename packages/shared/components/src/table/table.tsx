import React from "react"
import { Header } from "./header/header"
import { Preview } from "./cells/preview/preview"

export interface TableProps {
  columnNames: string[]
  cells: React.ReactNode[][]
}

export const Table = ({ columnNames, cells }: TableProps) => {
  const previewCells = [0, 1, 2, 3, 4].map((_) => columnNames.map((_) => <Preview />))

  const usedCells = cells.length > 0 ? cells : previewCells

  return (
    <div className="flex flex-col">
      <div className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
        <div className="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8">
          <div className="overflow-hidden border-b border-gray-200 shadow sm:rounded">
            <table className="min-w-full ">
              {<Header labels={columnNames} />}
              <tbody className="bg-white divide-y divide-gray-200">
                {usedCells.map((row, rowID) => {
                  return (
                    <tr key={rowID}>
                      {row.map((cell, colID) => {
                        return <td key={colID}>{cell}</td>
                      })}
                    </tr>
                  )
                })}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Table
