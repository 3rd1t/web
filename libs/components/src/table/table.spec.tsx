import { render } from "@testing-library/react";
import * as stories from "./table.stories";

describe("Table", () => {
  describe("Stories", () => {
    Object.entries(stories).forEach((s) => {
      if (s[0] !== "default") {
        it(s[0] + "should match snapshot", () => {
          const { baseElement } = render(s[1]());
          expect(baseElement).toMatchSnapshot();
        });
      }
    });
  });
});
