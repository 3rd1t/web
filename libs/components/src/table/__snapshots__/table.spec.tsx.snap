// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Stories multipleRowsshould match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col "
    >
      <div
        class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
      >
        <div
          class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
        >
          <div
            class="overflow-hidden border border-gray-200 sm:rounded"
          >
            <table
              class="min-w-full "
            >
              <thead>
                <tr>
                  <th
                    class="
                text-left
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Name
                  </th>
                  <th
                    class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Number
                  </th>
                  <th
                    class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Updated at
                  </th>
                </tr>
              </thead>
              <tbody
                class="bg-white divide-y divide-gray-200"
              >
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        Andreas Thomas
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex px-2 text-sm rounded-full text-purple-800 bg-purple-100"
                      >
                        123456789
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        22.09.2020
                      </span>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-lg font-medium text-gray-900"
                      >
                        Nico Webersinke
                      </span>
                      <p
                        class="text-sm text-gray-500"
                      >
                        Product lead
                      </p>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        123456789
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        22.09.2020
                      </span>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        Mads Jordt
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        123456789
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        22.09.2020
                      </span>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        Kevin Kohler
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        123456789
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        22.09.2020
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Table Stories onDarkshould match snapshot 1`] = `
<body>
  <div>
    <div
      class="p-10 bg-gray-900"
    >
      <div
        class="flex flex-col "
      >
        <div
          class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
        >
          <div
            class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
          >
            <div
              class="overflow-hidden border border-gray-200 sm:rounded"
            >
              <table
                class="min-w-full "
              >
                <thead>
                  <tr>
                    <th
                      class="
                text-left
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                    >
                      Name
                    </th>
                    <th
                      class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                    >
                      Number
                    </th>
                    <th
                      class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                    >
                      Updated at
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="bg-white divide-y divide-gray-200"
                >
                  <tr>
                    <td
                      class="text-left"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-sm leading-5 text-right text-gray-900"
                        >
                          Andreas Thomas
                        </span>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="inline-flex px-2 text-sm rounded-full text-purple-800 bg-purple-100"
                        >
                          123456789
                        </span>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-sm leading-5 text-right text-gray-900"
                        >
                          22.09.2020
                        </span>
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="text-left"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-lg font-medium text-gray-900"
                        >
                          Nico Webersinke
                        </span>
                        <p
                          class="text-sm text-gray-500"
                        >
                          Product lead
                        </p>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-sm leading-5 text-right text-gray-900"
                        >
                          123456789
                        </span>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <div
                          class="flex items-center justify-center"
                        >
                          <div
                            class="flex-shrink-0 w-10 h-10"
                          >
                            <div
                              class="text-red-800 bg-red-200"
                            >
                              <span>
                                <svg
                                  class="w-full"
                                  fill="none"
                                  stroke="currentColor"
                                  viewBox="0 0 24 24"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12"
                                    stroke-linecap="round"
                                    stroke-linejoin="round"
                                    stroke-width="2"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                          <div
                            class="flex flex-col items-start pl-2"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="text-left"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-sm leading-5 text-right text-gray-900"
                        >
                          Mads Jordt
                        </span>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-lg font-medium text-gray-900"
                        >
                          Lorem
                        </span>
                        <p
                          class="text-sm text-gray-500"
                        >
                          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                        </p>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-sm leading-5 text-right text-gray-900"
                        >
                          22.09.2020
                        </span>
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="text-left"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-sm leading-5 text-right text-gray-900"
                        >
                          Kevin Kohler
                        </span>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <span
                          class="text-sm leading-5 text-right text-gray-900"
                        >
                          123456789
                        </span>
                      </div>
                    </td>
                    <td
                      class="text-right"
                    >
                      <div
                        class="px-3 py-2 whitespace-no-wrap"
                      >
                        <button
                          class="text-sm font-medium leading-5 text-indigo-600 hover:text-indigo-900"
                        >
                          Click me
                        </button>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Table Stories previewModeshould match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col "
    >
      <div
        class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
      >
        <div
          class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
        >
          <div
            class="overflow-hidden border border-gray-200 sm:rounded"
          >
            <table
              class="min-w-full "
            >
              <thead>
                <tr>
                  <th
                    class="
                text-left
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Name
                  </th>
                  <th
                    class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Number
                  </th>
                  <th
                    class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Updated at
                  </th>
                </tr>
              </thead>
              <tbody
                class="bg-white divide-y divide-gray-200"
              >
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex w-full h-2 max-w-xs rounded-sm animate-pulse md:h-3 bg-gradient-to-t from-gray-200 to-gray-300"
                      />
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Table Stories singleRowshould match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col "
    >
      <div
        class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
      >
        <div
          class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
        >
          <div
            class="overflow-hidden border border-gray-200 sm:rounded"
          >
            <table
              class="min-w-full "
            >
              <thead>
                <tr>
                  <th
                    class="
                text-left
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Name
                  </th>
                  <th
                    class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Number
                  </th>
                  <th
                    class="
                text-right
                px-6 py-4 text-xs font-medium leading-4 tracking-wider text-gray-700 uppercase bg-gray-100"
                  >
                    Updated at
                  </th>
                </tr>
              </thead>
              <tbody
                class="bg-white divide-y divide-gray-200"
              >
                <tr>
                  <td
                    class="text-left"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        Andreas Thomas
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="inline-flex px-2 text-sm rounded-full text-purple-800 bg-purple-100"
                      >
                        123456789
                      </span>
                    </div>
                  </td>
                  <td
                    class="text-right"
                  >
                    <div
                      class="px-3 py-2 whitespace-no-wrap"
                    >
                      <span
                        class="text-sm leading-5 text-right text-gray-900"
                      >
                        22.09.2020
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
